vavr/src/main/java/io/vavr/Value.java

```diff
@@ -735,26 +856,69 @@ default CharSeq toCharSeq() {
     /**
      * Converts this to an {@link java.util.Optional}.
      *
+     * <pre>{@code
+     * // = Optional.empty
+     * Future.of(() -> { throw new Error(); })
+     *       .toJavaOptional()
+     *
+     * // = Optional[ok]
+     * Try.of(() -> "ok")
+     *     .toJavaOptional()
+     *
+     * // = Optional[1]
+     * List.of(1, 2, 3)
+     *     .toJavaOptional()
+     * }</pre>
+     *
      * @return A new {@link java.util.Optional}.
      */
     default Optional<T> toJavaOptional() {
         return isEmpty() ? Optional.empty() : Optional.ofNullable(get());
     }
 
     /**
-     * Converts this to a {@link java.util.Set}.
+     * Converts this to a mutable {@link java.util.Set}.
+     * Elements are added by calling {@link java.util.Set#add(Object}.
+     *
+     * <pre>{@code
+     * // = []
+     * Future.of(() -> { throw new Error(); })
+     *       .toJavaSet()
+     * 
+     * // = [ok]
+     * Try.of(() -> "ok")
+     *     .toJavaSet()
+     *
+     * // = [1, 2, 3]
+     * List.of(1, 2, 3)
+     *     .toJavaSet()
+     * }</pre>
      *
      * @return A new {@link java.util.HashSet}.
      */
     default java.util.Set<T> toJavaSet() {
-        return ValueModule.toJavaCollection(this, java.util.HashSet::new);
+        return ValueModule.toJavaCollection(this, java.util.HashSet::new, 16);
```

Comment:
How did you arrive at this magic number? I am sure it's an improvement.

Reply:
Screenshot taken From the `HashSet` source:

<img width="553" alt="screen shot 2017-09-12 at 19 51 35" src="https://user-images.githubusercontent.com/743833/30340783-f6d3f0c0-97f3-11e7-844c-85acb1135bc7.png">

Screenshot taken From the `HashMap` source:

<img width="449" alt="screen shot 2017-09-12 at 19 52 02" src="https://user-images.githubusercontent.com/743833/30340801-01201892-97f4-11e7-999a-849d85e7bac6.png">